homeassistant/components/unifiprotect/media_source.py

```diff
@@ -0,0 +1,790 @@
+"""UniFi Protect media sources."""
+
+from __future__ import annotations
+
+import asyncio
+from datetime import date, datetime, timedelta, timezone
+from enum import Enum
+from typing import Any, cast
+
+import pytz
+from pyunifiprotect.data import Camera, Event, EventType, SmartDetectObjectType
+from pyunifiprotect.exceptions import NvrError
+from pyunifiprotect.utils import from_js_time
+
+from homeassistant.components.media_player.const import (
+    MEDIA_CLASS_DIRECTORY,
+    MEDIA_CLASS_IMAGE,
+    MEDIA_CLASS_VIDEO,
+)
+from homeassistant.components.media_player.errors import BrowseError
+from homeassistant.components.media_source.models import (
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN
+from .data import ProtectData
+from .views import async_generate_event_video_url, async_generate_thumbnail_url
+
+VIDEO_FORMAT = "video/mp4"
+
+
+class SimpleEventType(str, Enum):
+    """Enum to Camera Video events."""
+
+    ALL = "All Events"
+    RING = "Ring Events"
+    MOTION = "Motion Events"
+    SMART = "Smart Detections"
+
+    @classmethod
+    def get_from_name(cls, name: str) -> SimpleEventType:
+        """Get SimpleEventType from name."""
+
+        if name.lower() == "smart_detect":
+            return SimpleEventType.SMART
+
+        types: list[SimpleEventType] = list(cls)
+        for event_type in types:
+            if event_type.name.lower() == name.lower():
+                return event_type
+        raise ValueError("Invalid event_type")
+
+    @classmethod
+    def get_event_type(cls, event_type: SimpleEventType) -> EventType | None:
+        """Get UniFi Protect event type from SimpleEventType."""
+
+        if event_type == SimpleEventType.ALL:
+            return None
+        if event_type == SimpleEventType.RING:
+            return EventType.RING
+        if event_type == SimpleEventType.MOTION:
+            return EventType.MOTION
+        return EventType.SMART_DETECT
+
+
+async def async_get_media_source(hass: HomeAssistant) -> MediaSource:
+    """Set up UniFi Protect media source."""
+
+    data_sources: dict[str, ProtectData] = {}
+    for data in hass.data[DOMAIN].values():
+        if isinstance(data, ProtectData):
+            data_sources[data.api.bootstrap.nvr.id] = data
+
+    return ProtectMediaSource(hass, data_sources)
+
+
+@callback
+def _timezone(hass: HomeAssistant) -> timezone:
+    return cast(timezone, pytz.timezone(hass.config.time_zone))
+
+
+@callback
+def _localize_dt(hass: HomeAssistant, timestamp: datetime) -> datetime:
+    return timestamp.astimezone(_timezone(hass))
+
+
+@callback
+def _now(hass: HomeAssistant) -> datetime:
+    return _localize_dt(hass, datetime.utcnow())
```

Comment:
I think you can use `dt_util.now()` here
`from homeassistant.util import dt as dt_util`

Reply:
Sweet, that looks like it does the same thing. I did not realize that used the timezone from the config correctly.